src/octree/mod.rs

```diff
@@ -267,6 +376,7 @@ impl OnDiskOctree {
             }
         }
         PointsInBoxIterator {
+            octree_data_provider: &*self.data_provider,
```

Comment:
Is the `&*` needed because you go from a `Box` to a normal reference? It looks kind of odd to me...

Reply:
Yes, it's needed here, otherwise the compiler complains...